app/section/sections/mobile-operating-systems/ui.test.js

```diff
@@ -0,0 +1,192 @@
+import nock from 'nock';
+import { Selector, ClientFunction } from 'testcafe';
+import { ManifestProvider } from '../../../manifestProvider';
+import dashboardWithCompleteSections from '../../../../fixtures/dashboardWithCompleteSections.json';
+
+const sectionId = 'mobile-operating-systems';
+const sectionApiUrl = `/api/v1/Solutions/S100000-001/sections/${sectionId}`;
+
+const mobileOperatingSystemsMarketingData = {
+  'operating-systems': ['apple-ios', 'android', 'other'],
+  'operating-systems-description': 'Text for description',
+};
+
+const sectionManifest = new ManifestProvider().getSectionManifest(sectionId);
+
+const getLocation = ClientFunction(() => document.location.href);
+
+const mocks = (responseStatus, responseBody) => {
+  nock('http://localhost:8080')
+    .get(sectionApiUrl)
+    .reply(responseStatus, responseBody);
+};
+
+const pageSetup = async (t, responseStatus = 200, responseBody = {}) => {
+  mocks(responseStatus, responseBody);
+  await t.navigateTo(`http://localhost:1234/S100000-001/section/${sectionId}`);
+};
+
+fixture.only('Show mobile operating systems')
+  .afterEach(async (t) => {
+    const isDone = nock.isDone();
+    if (!isDone) {
+      nock.cleanAll();
+    }
+
+    await t.expect(isDone).ok('Not all nock interceptors were used!');
+  });
+
+test('should render the page title', async (t) => {
+  await pageSetup(t);
+  const title = Selector('[data-test-id="section-title"]');
+  await t
+    .expect(title.innerText).eql(sectionManifest.title);
+});
+
+test('should render main advice of section', async (t) => {
+  await pageSetup(t);
+  const mainAdvice = Selector('[data-test-id="section-main-advice"]');
+  await t
+    .expect(mainAdvice.innerText).eql(sectionManifest.mainAdvice);
+});
+
+test('should render all the advice of the section', async (t) => {
+  await pageSetup(t);
+  const additionalAdvice = Selector('[data-test-id="section-additional-advice"]');
+  const expectedAdditionalAdvice = sectionManifest.additionalAdvice.join('\n\n');
+  await t
+    .expect(additionalAdvice.innerText).eql(expectedAdditionalAdvice);
+});
+
+test('should render the operating systems question', async (t) => {
+  const question = 'operating-systems';
+  const expectedOperatingSystems = sectionManifest.questions[question];
+  await pageSetup(t);
+  const opertingSystemsQuestion = Selector(`[data-test-id="question-${question}"]`);
+  await t
+    .expect(opertingSystemsQuestion.find('.nhsuk-fieldset__legend').innerText).eql(expectedOperatingSystems.mainAdvice)
+    .expect(opertingSystemsQuestion.find('.nhsuk-hint').innerText).eql(expectedOperatingSystems.additionalAdvice)
+    .expect(opertingSystemsQuestion.find('.nhsuk-checkboxes').count).eql(1)
+    .expect(opertingSystemsQuestion.find('.nhsuk-checkboxes__item').count).eql(3);
```

Comment:
Should we be checking the content here? Make sure the text of each checkbox it what we expect?

Reply:
yeah I can add this in (wasn't in the ui test I referenced)

Reply:
it isn't checked in client-application-types or browsers supported so they need to be updated then